src/poetry/console/exceptions.py

```diff
@@ -1,11 +1,224 @@
 from __future__ import annotations
 
+import dataclasses
+import shlex
+
+from dataclasses import InitVar
+from subprocess import CalledProcessError
+from typing import TYPE_CHECKING
+
 from cleo.exceptions import CleoError
 
+from poetry.utils._compat import decode
+
+
+if TYPE_CHECKING:
+    from cleo.io.io import IO
+
 
 class PoetryConsoleError(CleoError):
     pass
 
 
 class GroupNotFoundError(PoetryConsoleError):
     pass
+
+
+@dataclasses.dataclass
+class ConsoleMessage:
```

Comment:
I feel like there should be some basic tests for these classes since they contain quite a bit of logic.

Reply:
Probably, this came about as I was building the use cases. Honestly, just forgot I wanted to write them. Will add some unit tests.